arduino_mkrzero/src/lib.rs

```diff
@@ -0,0 +1,130 @@
+#![no_std]
+
+extern crate atsamd21_hal as hal;
+
+#[cfg(feature = "rt")]
+extern crate cortex_m_rt;
+#[cfg(feature = "rt")]
+pub use cortex_m_rt::entry;
+
+#[cfg(feature = "panic_abort")]
+pub extern crate panic_abort;
+
+pub use hal::atsamd21g18a::*;
+use hal::prelude::*;
+pub use hal::*;
+
+use gpio::{Floating, Input, Port};
+
+// This could possibly be simplified with concat_ident p for the name and P for the type
+// but I couldn't get it to work. Meta attribues (i.e. doc comments) are optional.
+macro_rules! input_pins {
+    // Insert trailing comma if it's missing
+    ($( $(#[$attr:meta])* pin $name:ident = $pin_name:ident: $pin_type:ident),+ ) =>
```

Comment:
nice macro!   FWIW, I would have (and have actually done that in other macros in this repo) just require the trailing `,`.  I think the audience for this macro is small enough that this isn't a terrible burden.

I'd be in favor of moving this into the `hal` crate and exporting it to the board support crates.

Reply:
I also wonder whether we can use something similar to this in `hal/src/gpio.rs` itself; I'm thinking of the lame/missing doc comments there.   I do wish that rust had a better story around constructing identifiers and strings in macros.